File: apps/api/presentation/restapi/kds_device_transformer.go
package restapi

import (
	"apps/api/domain"
	"encoding/json"
	apiContract "libs/api-contract"
	"net/http"
	"strconv"

	"github.com/gorilla/mux"
)

func GetKdsDeviceId(r *http.Request) (int64, error) {
	vars := mux.Vars(r)
	idParam := vars["kdsDeviceId"]
	id, err := strconv.ParseInt(idParam, 10, 32)
	return id, err
}

func GetKdsDeviceRequest(r *http.Request) (apiContract.KdsDeviceRequest, error) {
	var kdsDeviceRequest apiContract.KdsDeviceRequest
	err := json.NewDecoder(r.Body).Decode(&kdsDeviceRequest)
	return kdsDeviceRequest, err
}

func ToApiKdsDevice(device domain.KdsDevice) apiContract.KdsDevice {
	return apiContract.KdsDevice{
		Id:         device.Id,
		Name:       device.Name,
		PushToken:  device.PushToken,
		Platform:   string(device.Platform),
		LastSeenAt: device.LastSeenAt,
		CreatedAt:  device.CreatedAt,
		DeletedAt:  device.DeletedAt,
	}
}

func ToKdsDevice(kdsDeviceRequest apiContract.KdsDeviceRequest) domain.KdsDevice {
	return domain.KdsDevice{
		Name:      kdsDeviceRequest.Name,
		PushToken: kdsDeviceRequest.PushToken,
		Platform:  domain.KdsPlatform(kdsDeviceRequest.Platform),
	}
}

File: libs/ui/src/data/api/kdsDevice.transformer.ts
// eslint-disable-next-line @nx/enforce-module-boundaries
import { KdsDevice as ApiKdsDevice } from '../../../../api-contract/src';
import { KdsDevice, KdsDeviceRegisterParams } from '../../domain';

export function toKdsDevice(kdsDevice: ApiKdsDevice): KdsDevice {
  return {
    id: kdsDevice.id,
    name: kdsDevice.name,
    pushToken: kdsDevice.pushToken,
    platform: kdsDevice.platform,
    lastSeenAt: kdsDevice.lastSeenAt,
    createdAt: kdsDevice.createdAt,
    deletedAt: kdsDevice.deletedAt,
  };
}

export function toApiKdsDeviceRequest(params: KdsDeviceRegisterParams) {
  return {
    name: params.name,
    pushToken: params.pushToken,
    platform: params.platform,
  };
}
